tests/events/propagation.cpp

```diff
@@ -637,6 +647,110 @@ void EventPropagationTestCase::DocView()
 #endif // wxUSE_TOOLBAR
 }
 
+#if wxUSE_AUI
+    void EventPropagationTestCase::DocViewAui()
+    {
+        // Set up the parent frame and its menu bar.
+        wxDocManager docManager;
+
+        std::unique_ptr<wxDocParentFrameAny<wxAuiMDIParentFrame>>
+            parent(new wxDocParentFrameAny<wxAuiMDIParentFrame>(&docManager, nullptr, wxID_ANY, "Parent"));
+        wxAuiManager auiMgr(parent.get());
+
+        wxMenu* const menu = CreateTestMenu(parent.get());
+
+
+        // Set up the event handlers.
+        TestEvtSink sinkDM('m');
+        docManager.Connect(wxEVT_MENU,
+                           wxEventHandler(TestEvtSink::Handle), nullptr, &sinkDM);
+
+        TestEvtSink sinkParent('p');
+        parent->Connect(wxEVT_MENU,
+                        wxEventHandler(TestEvtSink::Handle), nullptr, &sinkParent);
+
+
+        // Check that wxDocManager and wxFrame get the event in order.
+        ASSERT_MENU_EVENT_RESULT( menu, "ampA" );
+
+
+        // The document template must be heap-allocated as wxDocManager owns it.
+        wxDocTemplate* const docTemplate = new wxDocTemplate
+                                               (
+                                                &docManager, "Test", "", "", "",
+                                                "Test Document", "Test View",
+                                                wxCLASSINFO(EventTestDocument),
+                                                wxCLASSINFO(EventTestView)
+                                               );
+
+        // Now check what happens if we have an active document.
+        wxDocument* const doc = docTemplate->CreateDocument("");
+        wxView* const view = doc->GetFirstView();
+
+        std::unique_ptr<wxDocChildFrameAny<wxAuiMDIChildFrame, wxAuiMDIParentFrame>>
+            child(new wxDocChildFrameAny<wxAuiMDIChildFrame, wxAuiMDIParentFrame>(doc, view, parent.get(), wxID_ANY, "Child"));
+
+        wxMenu* const menuChild = CreateTestMenu(child.get());
+
+        // Ensure that the child that we've just created is the active one.
+        child->Activate();
+
+    #ifdef __WXGTK__
```

Comment:
Is this hack/comment relevant for AUI?

Reply:
Based on experimentation, it appears not to be required.  I have put a `CHECK()` in my current version of the test to ensure it stays unnecessary.

For what it's worth, it seems a little strange to me that the hack is required for `wxDocMDIChildFrame`, and even stranger that it's required for `wxDocMDIChildFrame` but not for `wxAuiMDIChildFrame`, but not strange enough for me to bother trying to change that.